prql-query v0.0.14

pthariensflame opened the pull request:
Created with `brew bump-formula-pr`.

pthariensflame committed:
prql-query v0.0.14

snth commented:
@pthariensflame thank you for this.

Apologies, I don't use homebrew so for me it's difficult supporting this so any help or pointers would be appreciated.

Does this mean that for every release of prql-query this needs to be updated? What is that SHA compiled from?

pthariensflame commented:
@snth No worries!

The SHA is of the file specified in `url`, so it changes as the downloaded file does, naturally.  A change like this is needed every release of `prql-query` but it doesn't have to be manual; `prql-compiler`'s formula has been being updated by a GitHub Action, IIUC.

If you'd like, I can add to this PR by formatting and cleaning up the file to Homebrew's intended form.  ❤️

snth commented:
Thank you @pthariensflame . That would be amazing!

pthariensflame committed:
Fix and format both formulae

They now comply, at least nominally, with Homebrew standard style and recommendations.

This includes removing the explicit version line, since it's inferred from the URL successfully.  That required changes to the GitHub Action YAMLs.

pthariensflame committed:
Include Homebrew style audit in tests

pthariensflame commented:
I cleaned up both formulae and made the GitHub Action enforce style from now on.

snth commented:
Thank you very much @pthariensflame . Are you one of the homebrew maintainers/volunteers? Does this mean prql-query is now part of homebrew core like prql-query or it's still a tap/cask (apologies, I forgot the correct term and difference between the two)?

Also, would you say this is good to merge now?

I'm going to cut a prql-query release soon (today or tomorrow possibly) so we'll probably see soon if it all works.

pthariensflame commented:
This is still a separate tap, not “official”!  And no, I’m not part of Homebrew, just a knowledgeable volunteer.  😄  Something failed here so I want to investigate that before merging, but it should be good to go soon!

pthariensflame committed:
Fix prql-compiler test in GHAs

pthariensflame commented:
Okay; this should be good to merge as long as the tests pass!

snth commented:
Great, thanks so much!

snth reviewed (approved):
LGTM Thank you

max-sixty commented on .github/workflows/tests.yaml:
```diff
@@ -14,9 +14,10 @@ jobs:
       # prql-compiler
       - run: brew install prql/prql/prql-compiler
-      - run: echo "from employees | filter has_dog" | prql-compiler compile
+      - run: echo "from employees | filter has_dog" | prqlc compile
```
Great, thanks for changing this

max-sixty replied:
FYI I adjusted this in `main` to stop it breaking.

@snth please can you review the `prql-query` part of this so we can merge this PR?

max-sixty replied:
Oh, I guess this was the approval? https://github.com/PRQL/homebrew-prql/pull/10#pullrequestreview-1262596083

Sorry, so maybe you were waiting on me!!

I'll try and merge main back in and then merge this. Sorry for the delay @pthariensflame .

pthariensflame replied:
No need to worry!  I've been held up too.

max-sixty commented on Formula/prql-compiler.rb:
```diff
@@ -1,19 +1,17 @@
 class PrqlCompiler < Formula
-  desc "PRQL is a modern language for transforming data — a simpler and more powerful SQL"
+  desc "PRQL is a modern language for transforming data: a simpler and more powerful SQL"
```
Not a huge deal, but the existing line is a canonical sentence we use in lots of places; do we have a strong reason to change it? If so, we should change it everywhere...

pthariensflame replied:
I only changed it to avoid hitting the length limit Homebrew prefers for descriptions.  Is there a better way to do that you'd prefer?

max-sixty replied:
Oh interesting! That's fine then. I guess we could shorten "language" to "lang" but I probably prefer your change

max-sixty commented on Formula/prql-compiler.rb:
```diff
@@ -1,19 +1,17 @@
 class PrqlCompiler < Formula
```
Do you think we could change the name of this file to `prqlc.rb`? Ideally the recent change should mean we use `prql/prql/prqlc` rather than `prql/prql/prql-compiler` (and I don't know whether it's possible to deprecate the old one or we just remove it, fine to remove if it that's easier...)

pthariensflame replied:
It's entirely possible to both change the canonical formula name and keep the old name around as an alias!  I can make that change in a few hours when I get home.

max-sixty replied:
Super! Thank you

pthariensflame replied:
Sorry about so long a delay!  I'll make it happen in another PR.

max-sixty reviewed (commented):
Thanks a lot for spotting this @pthariensflame and making the changes.

I left a couple of comments around the `prql-compiler` changes, let me know what you think.

max-sixty committed:
Merge branch 'main' into bump-prql-query-v0.0.14

max-sixty committed:
find replace prql-compiler -> prqlc

max-sixty commented:
Because of https://github.com/PRQL/homebrew-prql/issues/11, I'm going to merge and then we can iterate...

snth commented:
Thank you
